File: frontend/lib/features/recipe/models/recipe_ingredient.dart
//one ingredient line in a recipe, mirrors recipe_ingredients table the new update database table
class RecipeIngredient {
  final int ingredientId;
  final int recipeId;
  final int ingId;
  final String? name;
  final double? quantity;
  final String? unit;
  final int sortOrder;

  const RecipeIngredient({
    required this.ingredientId,
    required this.recipeId,
    required this.ingId,
    this.name,
    this.quantity,
    this.unit,
    this.sortOrder = 0,
  });

  factory RecipeIngredient.fromJson(Map<String, dynamic> json) {
    return RecipeIngredient(
      ingredientId: json['ingredient_id'] as int,
      recipeId: json['recipe_id'] as int,
      ingId: json['ing_id'] as int,
      name: json['ingName'] as String?,
      quantity: (json['quantity'] as num?)?.toDouble(),
      unit: json['unit'] as String?,
      sortOrder: (json['sort_order'] as int?) ?? 0,
    );
  }
  Map<String, dynamic> toJson() => {
        'ingId': ingId,
        'quantity': quantity,
        'unit': unit,
        'sortOrder': sortOrder,
      };
}

File: backend/src/main/java/com/mealchemy/recipe/model/RecipeIngredient.java
package com.mealchemy.recipe.model;

/* Import libraries */

import jakarta.persistence.*;

/* Import classes */

@Entity
@Table(name = recipe_ingredients)
public class RecipeIngredient
{
    /* Declaring fields */
    @Id 
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ingredient_id")
    private int ingredientId;
    
    @Column(name = "recipe_id", nullable = false)
    private int recipeId;

    @Column(name = "ing_id", nullable = false)
    private int ingId;

    @Column(name = "quantity", precision = 10, scale = 3, nullable = false)
    private BigDecimal quantity;

    @Column(name = "unit", length = 30, nullable = false)
    private String unit;

    @Column(name = "sort_order", nullable = false)
    private int sortOrder;  

    /* Getters */

    public int getIngredientId()
    {
        return ingredientId;
    }

    public int getRecipeId()
    {
        return recipeId;
    }

    public int getIngId()
    {
        return ingId;
    }

    public BigDecimal getQuantity()
    {
        return quantity;
    }

    public String getUnit()
    {
        return unit;
    }

    public int getSortOrder()
    {
        return sortOrder;
    }

    /* Setters */
}
